### ISSUE TITLE
install ess from apt and use R built from source

### ISSUE THREAD

**Issue by jangorecki** [emacs-ess/ESS#579]
I would like to install emacs ess in convenient way from apt repository using `sudo apt install emacs ess`.
I use R built from sources, not installed from apt, thus apt also wants to install R from apt as it is dependency for ess:
```
The following additional packages will be installed:
  ... r-base-core r-base-dev ...
```
Is there any way to tell installation process that I have R already installed so I don't have to install R from apt?

**Comment by lionel-** [comment]
Please ask the package maintainers of your distribution.

**Comment by lionel-** [comment]
oh or perhaps on the ESS-help mailing list.

**Comment by vspinu** [comment]
Looks like you are on a wrong list. R is not a dependency of ESS and should not be required when you install ESS.

**Comment by eddelbuettel** [comment]
@jangorecki What exectly is your issue?  As the Debian maintainer I am happy to listen to specifics about the package.  _Not_ depending on R makes no sense for the distribution as you install ESS (generally) to be used with R.  Your case is a very special case.

[ There is a means to "trick" `apt` et al into thinking you have a package when you don't -- an old helper package allowed to create an essentially "empty" package to satisfy the depend. You could look into that.  Or create your own R package, or, just install r-base-dev and r-base-core. ]

**Comment by jangorecki** [comment]
@eddelbuettel Issue is to install ess from apt but not do redundant stuff, like installing R from apt, which I am not planning to use. I can imaging other use cases when user want to use remote R session over ssh with local emacs, so there is no R locally available, even built from source. But trick you wrote is too much hassle, it seems that answer to my question is "there is no simple way, just install R from apt".

**Comment by lionel-** [comment]
You can also install ESS from melpa, which is very easy nowadays.

**Comment by eddelbuettel** [comment]
There is no "redundant stuff".  It is called a dependency, and it is there for a reason.  But this is _entirely the wrong forum for this_ so if you insist on continuing this somewhat inapplicable discussion please do it as a Debian bug report.  I still think I can't do anything for you there but ...

**Comment by eddelbuettel** [comment]
As discussed here, acted upon (ie issue closed) and just discussed in #671, this not the right place for this discussion.

I as Debian maintainer have the discretion to set a package up to my best judgement.  In all those years of maintaining ess, I have yet to encounter a non-R user.  As such, Jan's request is a corner case that we overrule to have sane default behaviour.  People install ess for R. 

And all this is moot anyway.  If you don't want the package, use `M-x package-install`.  No different from installing either `r-cran-data.table` as binary, or from source (which willl Build-Depends) as will using `install.packages()` to install from CRAN.

**Comment by vspinu** [comment]
> this not the right place for this discussion.

Methods of package distribution are of core interests of the package developers and maintainers. IMO this is a more appropriate place than the deb list because all core developers can chime in. 

> > I as Debian maintainer have the discretion to set a package up to my best judgement.

Well, ok. And does it make a huge difference to you if you move `r-base` from "Recommended" to "Suggested"?

> And all this is moot anyway. If you don't want the package, use M-x package-install

Well, there is an easy to overlook elephant in the room. People might intentionally avoid MELPA for security reasons. It's inherently insecure. Otherwise there wouldn't be much point in distro distribution indeed.

**Comment by dickmao** [comment]
> As discussed here, acted upon (ie issue closed) and just discussed in #671, this not the right place for this discussion.

Then where is the correct place?  Is it https://salsa.debian.org/edd/ess
I created a `dickmao-guest` account, but don't seem to have the correct privileges to open an issue.

**Comment by eddelbuettel** [comment]
What is your question?  Debian bug reports go to the Debian BTS at [bugs.debian.org](http://bugs.debian.org) (though email submission is preferred, there is even an emacs (minor-)mode.

I don't think salsa is meant to act as a GitHub replacement to collect submissions and issues.

**Comment by dickmao** [comment]
my question seems far too specific to ESS to warrant posting to `bugs.debian.org` and `r-sig-debian`.  here it is: 

If we ensure the `tgz` from `debian/watch` is self-contained (contains all the docs that went missing in #731), then does `debian/rules` need to mention `make` at all?  And if not, then why had mmaechler admonished so strongly against changing `Makefile` and `Makeconf` in #747?

**Comment by eddelbuettel** [comment]
If I understand correctly what you are asking then this is circular.  You are questioning how Debian (via `uscan` and other tools) gets tarballs from upstream, and/or how Debian assumes (or even tests) if the tarball is complete.

And it is just that. We assume we can trust upstream, and that upstream is not breaking things release to release.  And if and when things break then we change the scripts consuming the content (as there is really no other choice ...).  

But that assumption and what is behind it is the issue that went astray here.  And that change was inside the ess tarball / source as a change in process was implemented.  It was not a change at the Debian side.  It's much simpler.  If `configure; make; make install` (plus maybe `make doc` or `make check` or ...) works, why change it?    But that is what happened here.  

So in short, and to sum up:  I am still not entirely sure why you opened this issue from June.

**Comment by mmaechler** [comment]
> ............ then does `debian/rules` need to mention `make` at all? And if not, then why had mmaechler admonished so strongly against changing `Makefile` and `Makeconf` in #747?

That was not related to Debian.  Problem were reported by the Windows and SuSE Linux packagers.
